extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -179,9 +187,27 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
 
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
-
-                return getActiveConnections(a).size()
-                     - getActiveConnections(b).size();
+                
+                int weightA, weightB;
+                // Check if weight of a is non-null and retrieve it.
+                if (a.getConnectionWeight() != null && a.getConnectionWeight().intValue() > 0)
+                    weightA = a.getConnectionWeight().intValue();
+                // In all other cases assign 1 for sorting.
+                else
+                    weightA = 1;
+
+                // Check if weight of b is null, assign 1 if it is.
+                if (b.getConnectionWeight() != null && b.getConnectionWeight().intValue() > 0)
+                    weightB = b.getConnectionWeight().intValue();
+                // In all other cases assign 1 for sorting.
+                else
+                    weightB = 1;
+
+                // Get current active connections, add 1 to both to avoid calculations with 0.
+                int connsA = getActiveConnections(a).size() + 1;
+                int connsB = getActiveConnections(b).size() + 1;
+
+                return (connsA * 10000 / weightA) - (connsB * 10000 / weightB);
```

Comment:
Yes, clearly not the best code ever written.

> With normal WLC, the idea is that a server with weight N should be allocated N times as many connections as a server with weight 1, correct?

Well, I'd say that the idea of weight in general is as you say - but the WLC also factors in the number of current connections.  There's also WRR, which accounts for the last time a system was used for a connection combined with weight.  But, yes, that's the overall idea...

> return (connsA * weightB) - (connsB * weightA);

Works for me, I'll update the code.  I agree that the actual value that is returned or computed does not matter - the sign and the ability to get a meaningful integer value are really all that I was going for, and that seems to solve it more elegantly :-).  I'll update the code, test a few times and commit the changes.

Thanks for the help!